bazel/repositories.bzl

```diff
@@ -140,12 +161,23 @@ def _go_deps(skip_targets):
         _repository_impl("io_bazel_rules_go")
 
 def _envoy_api_deps():
-    _repository_impl("envoy_api")
+    # Treat Envoy's overall build config as an external repo, so projects that
+    # build Envoy as a subcomponent can easily override the config.
+    if "envoy_api" not in native.existing_rules().keys():
+        _default_envoy_api(name="envoy_api")
 
     native.bind(
         name = "http_api_protos",
         actual = "@googleapis//:http_api_protos",
     )
+    _repository_impl(
```

Comment:
Why these changes out of curiosity?

Reply:
I think these were needed to run the API tests, which we previously only did from data-plane-api (where the equivalent is in the WORKSPACE).